getsentry/sentry
# bug(replays): Fix loading and paging through the Issues>Replays tab

Basically a few things are happening:
1. Call `useLocation` less often
  Because sometimes we're going to pass in our own value instead of using the real location.

2. Don't fetch replayIds inside static/app/views/organizationGroupDetails/groupDetails.tsx anymore.
  We used to do it in here because there was some code-sharing between the count that appears in the tab, and the list contents. That's fixed now -> the tab loads it's own count now. Now we can delegate fetching data only when the specific tab is open, and drop the `this.getCurrentTab() === Tab.REPLAYS` checks. (which had issues)

3. `<GroupReplays>` now has a new child: `<GroupReplaysTable>`
  Its is mostly there so I can do `return null` in the parent and conditionally call `useReplayList()` in the child
  All the replayIds fetching logic that was previously inside groupDetails.tsx is moved, and improved, into `<GroupReplays>`
   

**Test Plan:**
1. Visit the Issues>Replay tab
2. Click from another Issues tab into Replays tab
3. Click from All Events (page 2 or beyond) into Replays tab


- Notice the tab loads
- No flash & re-fetching/re-render of content
- Pagination works

Fixes #39464
Fixes #41836
